arch/risc-v/src/common/espressif/esp_wlan.h

```diff
@@ -0,0 +1,184 @@
+/****************************************************************************
+ * arch/risc-v/src/common/espressif/esp_wlan.h
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+#ifndef __ARCH_RISCV_SRC_COMMON_ESPRESSIF_ESP_WLAN_H
+#define __ARCH_RISCV_SRC_COMMON_ESPRESSIF_ESP_WLAN_H
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#ifndef __ASSEMBLY__
+
+#undef EXTERN
+#if defined(__cplusplus)
+#define EXTERN extern "C"
+extern "C"
+{
+#else
+#define EXTERN extern
+#endif
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#if defined(CONFIG_ESPRESSIF_WIFI_STATION)
+#  define ESP_WLAN_HAS_STA
+#  define ESP_WLAN_STA_DEVNO    0
+#  define ESP_WLAN_DEVS         1
+#elif defined(CONFIG_ESPRESSIF_WIFI_SOFTAP)
+#  define ESP_WLAN_HAS_SOFTAP
+#  define ESP_WLAN_SOFTAP_DEVNO 0
+#  define ESP_WLAN_DEVS         1
+#elif defined(CONFIG_ESPRESSIF_WIFI_STATION_SOFTAP)
+#  define ESP_WLAN_HAS_STA
+#  define ESP_WLAN_HAS_SOFTAP
+#  define ESP_WLAN_STA_DEVNO    0
+#  define ESP_WLAN_SOFTAP_DEVNO 1
+#  define ESP_WLAN_DEVS         2
+#endif
+
+#define MAC_LEN                 (6)
+
+#ifdef CONFIG_ESPRESSIF_WIFI
+
+/****************************************************************************
+ * Inline Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: nuttx_err_to_freertos
+ *
+ * Description:
+ *   Transform from Nuttx OS error code to FreeRTOS's pdTRUE or pdFALSE.
+ *
+ * Input Parameters:
+ *   ret - NuttX error code
+ *
+ * Returned Value:
+ *   Wi-Fi adapter error code
+ *
+ ****************************************************************************/
+
+static inline int32_t nuttx_err_to_freertos(int ret)
```

Comment:
```suggestion
static inline bool nuttx_err_to_freertos(int ret)
```
or `pdTRUE`/`pdFALSE` are base typed values?

Reply:
Yes, they are [`BaseType_t`](https://github.com/espressif/esp-idf/blob/9da5d7c40a437e6a33fd7ad68573c2dd2ae26c0b/components/freertos/FreeRTOS-Kernel-SMP/include/freertos/projdefs.h#L43). That's why I preferred to keep it as `int32_t`.

Reply:
Most of the comments are just a recommendation. Let's keep `int32_t`